src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -1,39 +1,16 @@
 import { Canvas } from 'react-three-fiber';
-import React, { useEffect, useMemo } from 'react';
-import { computeTextureData } from './utils';
+import React from 'react';
 import styles from './HeatmapVis.module.css';
 import ColorBar from './ColorBar';
-import { useHeatmapStore, selectDataScale, selectColorScale } from './store';
-import { useHeatmapStyles } from './hooks';
+import { useHeatmapStyles, useProps } from './hooks';
 import AxisGrid from './AxisGrid';
 import Mesh from './Mesh';
 import Tooltip from './Tooltip';
+import HeatmapProvider from './HeatmapProvider';
 
-const AXIS_OFFSETS: [number, number] = [72, 36];
-
-interface Props {
-  dims: [number, number];
-  data: number[][];
-}
-
-function HeatmapVis(props: Props): JSX.Element {
-  const { dims, data } = props;
-
-  const findDomain = useHeatmapStore(state => state.findDomain);
-  const colorScale = useHeatmapStore(selectColorScale);
-  const dataScale = useHeatmapStore(selectDataScale);
-
-  const [mapAreaRef, heatmapStyles] = useHeatmapStyles(dims, AXIS_OFFSETS);
-
-  const values = useMemo(() => data.flat(), [data]);
-  const textureData = useMemo(
-    () => computeTextureData(values, colorScale, dataScale),
-    [dataScale, colorScale, values]
-  );
-
-  useEffect(() => {
-    findDomain(values);
-  }, [findDomain, values]);
```

Comment:
Yep, `HeatmapVis` doesn't do much anymore... 😄 Each child component takes care of retrieving what it needs by itself.